hugr-core/src/extension/op_def.rs

```diff
@@ -117,26 +118,17 @@ pub trait CustomLowerFunc: Send + Sync {
 /// Encode a signature as `PolyFuncTypeRV` but optionally allow validating type
 /// arguments via a custom binary. The binary cannot be serialized so will be
 /// lost over a serialization round-trip.
-#[derive(serde::Deserialize, serde::Serialize)]
 pub struct CustomValidator {
-    #[serde(flatten)]
     poly_func: PolyFuncTypeRV,
-    #[serde(skip)]
+    //Optional custom function for validating type arguments before returning the signature.
```

Comment:
This doesn't look very optional ?

Reply:
Right, I'd started commenting in a different direction, but making this non-optional is a different way of doing what I was going to suggest, that works for me. Just correct the comment tho :)